documentation/sdkautomation/sdk_customization.md

```diff
@@ -0,0 +1,754 @@
+              "const": false
+            }
+          ],
+          "default": false,
+        },
+        "buildScript": {
+          // Build the generated sdk.
+          // Param: <path_to_package_folder>
+          // Package folder could be a list separated by space if it's from generateOutput.json.
+          "$ref": "#/definitions/RunOptions"
+        },
+        "changelogScript": {
+          // Changelog generation and breaking-change detection.
+          // Param: <path_to_package_folder>
+          // Package folder could be a list separated by space if it's from generateOutput.json.
+          // Expected output from stdout/stderr: Changelog in markdown
+          "allOf": {
+            "$ref": "#/definitions/RunOptions"
+          },
+          "properties": {
+            "breakingChangeDetect": {
+              // If stdout or stderr matches this in output of changelog tool
+              // then we assume this SDK has breaking change.
+              "$ref": "#/definitions/RunLogFilterOptions"
+            }
+          }
+        }
+      }
+    },
+    "artifactOptions": {
+      "artifactPathFromFileSearch": {
+        "oneOf": [
+          {
+            // If this option is set to object, then artifacts will be detected automatically
+            //   based on filename regex search.
+            // This options must be set to object if parseGenerateOutput is false.
+            "type": "object",
+            "properties": {
+              "searchRegex": {
+                // Any file under package folder matching the searchRegex is package artifact.
+                "type": "string",
+                "format": "regex",
+              }
+            }
+          },
+          {
+            // If this option is set to false, then package folder will be from generateOutput.json
+            "const": false
+          }
+        ],
+        "default": false
+      },
+      "installInstructionScript": {
+        // Generate install instruction that could be shown in spec PR comment (lite version)
+        //  or in generated SDK PR (full version).
+        // If generate_output.json contains install_instruction then this could be skipped.
+        // Param: <path_to_installInstructionInput.json> <path_to_installInstructionOutput.md> <lite|full>
+        // installInstructionInput.json: See #InstallInstructionScriptInput .
+        // installInstructionOutput.md: Install instruction in markdown.
+        "allOf": {
+          "$ref": "#/definitions/RunOptions"
+        },
+        "properties": {
+          "enableLiteInstallInstruction": {
+            "type": "boolean",
+            "default": false
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RunOptions": {
+      // Options to run a script and collect log.
+      "type": "object",
+      "properties": {
+        "path": {
+          // Script path related to repo root
+          "type": "string"
+        },
+        "logPrefix": {
+        },
```

Comment:
This property is missing a type I suppose?

Reply:
Fixed